### PR TITLE
Fix pixel interpolate method

### PR DESCRIPTION
**Pull request by dlemstra**
This PR addresses the following issues that should be addressed before the 4.0 release:

- `BicubicInterpolatePixel` has been deprecated and replaced with `CatromInterpolatePixel`.
- `FilterInterpolatePixel` has been removed and there is no replacement in IM7.
- `NearestNeighborInterpolatePixel` has been renamed to `NearestInterpolatePixel` in IM7.

### CODE REVIEW

File: ext/RMagick/rmmain.c [hunk 1]
```diff
@@ -1295,12 +1295,10 @@ Init_RMagick2(void)
     DEF_ENUM(PixelInterpolateMethod)
         ENUMERATOR(UndefinedInterpolatePixel)
         ENUMERATOR(AverageInterpolatePixel)
-        ENUMERATOR(BicubicInterpolatePixel)
         ENUMERATOR(BilinearInterpolatePixel)
-        ENUMERATOR(FilterInterpolatePixel)
```
Comment (mockdeep) [inline ext/RMagick/rmmain.c]: What happens to `BicubicInterpolatePixel` and `FilterInterpolatePixel`? Are these deprecated constants?

Comment (dlemstra) [reply ext/RMagick/rmmain.c]: This is described in the summary?

Comment (mockdeep) [reply ext/RMagick/rmmain.c]: Oh, sorry, I missed that. Okay, so these are removed. Then these should be part of the 4.0 release, right? Did we have deprecations for them? And for `NearestInterpolatePixel` we should have that as part of 3.2.1?

File: ext/RMagick/rmmain.c [hunk 2]
```diff
@@ -1295,12 +1295,10 @@ Init_RMagick2(void)
...
         ENUMERATOR(IntegerInterpolatePixel)
         ENUMERATOR(MeshInterpolatePixel)
-        ENUMERATOR(NearestNeighborInterpolatePixel)
+        ENUMERATORV(NearestInterpolatePixel, NearestNeighborInterpolatePixel)
```
Comment (mockdeep) [inline ext/RMagick/rmmain.c]: So are we deprecating `NearestNeighborInterpolatePixel`, or are we going to keep them aliased for a while?

File: ext/RMagick/rmmain.c (hunk 1 above)
Comment (dlemstra) [reply ext/RMagick/rmmain.c]: We should have put that in the 3.2.0 release but we missed it. I don't think we need to put out a separate release for this but just do it as a breaking change and release 4.0 after this has been merged.

File: ext/RMagick/rmmain.c (hunk 2 above)
Comment (dlemstra) [reply ext/RMagick/rmmain.c]: Not sure what you are asking. I am on Skype now if you want to chat.

Comment (mockdeep) [reply ext/RMagick/rmmain.c]: Let's talk tomorrow if you're available. I'm pretty busy today.

Comment (dlemstra) [reply ext/RMagick/rmmain.c]: That will work for me.

**Pull request merged**